### PR TITLE
Feat/ew 27 create user - sign up

### PR DESCRIPTION
**Pull request by vietminhhieu**
Try hard, not go to bed early

### CODE REVIEW

File: src/components/Snackbar/CustomSnackbar.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        anchorOrigin={{
+          vertical: 'top',
+          horizontal: 'center',
+        }}
+        open={open}
+        autoHideDuration={6000}
+        onClose={handleClose}
+        message="Note archived"
+      >
+        <Alert onClose={handleClose} severity="success" sx={{ width: '100%' }}>
```
Comment (Duc-Quang-0310) [inline src/components/Snackbar/CustomSnackbar.tsx:31]: Should assign severity = { variable } not always using success type. Implement AlertColor from MUI

File: src/pages/client/auth/passwordRecover/PasswordRecover.tsx [hunk 2]
```diff
@@ -1,5 +1,58 @@
...
 export const PasswordRecover: React.FC = () => {
-  return <div>Password Recover</div>;
+  const [loading, setLoading] = useState(false);
+
+  return (
+    <AuthForm imageSide={ImageSide.LEFT}>
+      <div className="password_recover-container">
+        <h1 className="password_recover-heading">Password Recover</h1>
+
+        <h3 className="password_recover-heading_h3">Send to Gara Auto Email Address</h3>
```
Comment (Duc-Quang-0310) [inline src/pages/client/auth/passwordRecover/PasswordRecover.tsx:21]: Remove this line of code

File: src/pages/client/auth/passwordRecover/PasswordRecover.tsx [hunk 3]
```diff
@@ -1,5 +1,58 @@
...
+
+        <h5 className="password_recover-heading_h5">Please enter your email address which you want to recover</h5>
```
Comment (Duc-Quang-0310) [inline src/pages/client/auth/passwordRecover/PasswordRecover.tsx:23]: Remove this line of code

File: src/pages/client/auth/signUp/SignUp.tsx [hunk 4]
```diff
@@ -1,26 +1,167 @@
...
+    delete signUpObj?.reTypePassword;
+
+    try {
+      setLoading(true);
+      const response = await clientService.userSignUp(signUpObj);
+      console.log('response: ', response);
+      setResponseFromAPI(response);
+      setShowSnackbar(true);
+    } catch (error: any) {
+      console.log('error: ', error.response?.data.message);
```
Comment (Duc-Quang-0310) [inline src/pages/client/auth/signUp/SignUp.tsx:41]: Remove console.log()

File: src/pages/client/auth/signUp/SignUp.tsx [hunk 5]
```diff
@@ -1,26 +1,167 @@
...
+      setResponseFromAPI(error.response?.data?.message);
```
Comment (Duc-Quang-0310) [inline src/pages/client/auth/signUp/SignUp.tsx:42]: Refer error.response to a type

File: src/reduxToolKit-Saga/auth/LoginSaga.ts [hunk 6]
```diff
@@ -33,7 +33,9 @@ function* loginSaga(action: PayloadAction<LoginParams>) {
       yield put(reset());
     }
   } catch (error: any) {
+    console.log('error', error.response);
```
Comment (Duc-Quang-0310) [inline src/reduxToolKit-Saga/auth/LoginSaga.ts:36]: remove this line of code

**Review by Duc-Quang-0310 — CHANGES_REQUESTED** [review]
Resolve comment before merge

File: src/components/Snackbar/CustomSnackbar.tsx (hunk 1 above)
Comment (Duc-Quang-0310) [reply src/components/Snackbar/CustomSnackbar.tsx:31]: image

File: src/pages/client/auth/signUp/SignUp.tsx (hunk 5 above)
Comment (vietminhhieu) [reply src/pages/client/auth/signUp/SignUp.tsx:42]: I don't understand this comment. Can you explain clear for me?

File: src/reduxToolKit-Saga/store.ts [hunk 7]
```diff
@@ -2,7 +2,7 @@ import createSagaMiddleware from '@redux-saga/core';
 import { persistStore } from 'redux-persist';
 import { combineReducers, configureStore, Reducer, Store } from '@reduxjs/toolkit';
 
-import rootSaga from './rootSaga';
+import rootSaga from './rootsaga';
```
Comment (Duc-Quang-0310) [inline src/reduxToolKit-Saga/store.ts:5]: import rootSaga from './rootSaga'

File: src/components/Snackbar/CustomSnackbar.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { Snackbar, Alert } from '@mui/material';
+
+interface SnackbarProps {
+  snackbarColor: any;
```
Comment (Duc-Quang-0310) [inline src/components/Snackbar/CustomSnackbar.tsx:5]: import { Snackbar, AlertColor } from "@mui/material";

 snackbarColor: AlertColor

**Review by Duc-Quang-0310 — CHANGES_REQUESTED** [review]

**Pull request merged**